addons/website_slides/static/src/js/public/components/slide_quiz_finish_dialog/slide_quiz_finish_dialog.js

```diff
@@ -0,0 +1,70 @@
+/** @odoo-module **/
+
+import { Component, onMounted, useState } from "@odoo/owl";
+import { browser } from "@web/core/browser/browser";
+import { Dialog } from "@web/core/dialog/dialog";
+import { SlideXPProgressBar } from "@website_slides/js/public/components/slide_quiz_finish_dialog/slide_xp_progress_bar";
+import { _t } from "@web/core/l10n/translation";
+
+export class SlideQuizFinishDialog extends Component {
+    static components = { Dialog, SlideXPProgressBar };
+    static props = {
+        close: Function,
+        hasNext: Boolean,
+        onClickNext: Function,
+        quiz: Object,
+        userId: Number,
+    };
+    static template = "website_slides.SlideQuizFinishDialog";
+
+    setup() {
+        super.setup();
+        this.state = useState({
+            animateKarmaGain: false,
+            fadeRankMotivational: false,
+            hideDismissBtns: true,
+            showRankMotivational: false,
+        });
+        this.title = _t("Amazing!");
+        onMounted(() => {
+            this._animateText();
+        });
+    }
+
+    //--------------------------------
+    // Handler
+    //--------------------------------
+
+    onClickNext() {
+        this.props.onClickNext();
+        this.props.close();
+    }
+
+    //--------------------------------
+    // Private
+    //--------------------------------
+
+    /**
+     * Handles the animation of the different text such as the karma gain
+     * and the motivational message when the user levels up.
+     * @private
+     */
+    _animateText() {
+        browser.setTimeout(() => {
+            this.state.animateKarmaGain = true;
+            this.state.hideDismissBtns = false;
+        }, 800);
+
+        if (this.props.quiz.rankProgress.level_up) {
+            this.title = _t("Level up!");
```

Comment:
Following your answer to my previous comment, shouldn't this conditional assignment title be done in the setup?